internal/usecase/interactor/auth.go

```diff
@@ -90,28 +115,19 @@ func NewAuthStorage(cfg *StorageConfig, request repo.AuthRequest, getUserBySubje
 	}
 }
 
-func initKeys(name pkix.Name) (*rsa.PrivateKey, jose.SigningKey, jose.JSONWebKeySet) {
+func initKeys(keyBytes, certBytes []byte) (*rsa.PrivateKey, jose.SigningKey, jose.JSONWebKeySet) {
 
-	key, err := rsa.GenerateKey(rand.Reader, 2048)
-	if err != nil {
-		panic(err)
+	block, _ := pem.Decode(keyBytes)
+	if block == nil {
+		panic("failed to create the key")
```

Comment:
```suggestion
	block, err := pem.Decode(keyBytes)
	if err != nil || block == nil {
		return nil, err
```

Reply:
`pem.Decode` dos not returns error as a second return value, check the docs [here](https://pkg.go.dev/encoding/pem#Decode)